ucm2/HDA/HiFi-mic.conf

```diff
@@ -0,0 +1,409 @@
+		If.justmic {
+			Condition {
+				Type ControlExists
+				Control "name='${var:SourceControl}'"
+				ControlEnum "Mic"
+			}
+			True.If.whichdev {
+				Condition {
+					Type String
+					Empty "${var:DeviceMicName}"
+				}
+				True.Define {
+					DeviceMicName "Mic"
+					DeviceMicComment "Stereo Microphone"
+					DeviceMicJack "Mic Jack"
+					DeviceMicPriority 400
+				}
+				False.If.whichdev2 {
+					Condition {
+						Type String
+						Empty "${var:DeviceMic2Name}"
+					}
+					True.Define {
+						DeviceMic2Name "Mic"
+						DeviceMic2Comment "Stereo Microphone"
+						DeviceMic2Jack "Mic Jack"
+						DeviceMic2Priority 400
+					}
+					False.If.whichdev3 {
+						Condition {
+							Type String
+							Empty "${var:DeviceMic2Name}"
+						}
+						True.Define {
+							DeviceMic3Name "Mic"
+							DeviceMic3Comment "Stereo Microphone"
+							DeviceMic3Jack "Mic Jack"
+							DeviceMic3Priority 400
+						}
+						False.Define {
+							DeviceMic4Name "Mic"
+							DeviceMic4Comment "Stereo Microphone"
+							DeviceMic4Jack "Mic Jack"
+							DeviceMic4Priority 400
+						}
+					}
+
+				}
+			}
+		}
+	}
+	True {
+		# the input selection is via switches, the possible types are
+		# Mic and Front Mic
+		Define {
+			DeviceMicName "Mic"
+			DeviceMicComment "Stereo Microphone"
+			DeviceMicJack "Mic Jack"
+			DeviceMicPriority 200
+		}
+
+		If.fmic {
+			Condition {
+				Type ControlExists
+				Control "name='Front Mic Playback Switch'"
+			}
+			True.Define {
+				DeviceMic2Name "Front Mic"
+				DeviceMic2Comment "Front Stereo Microphone"
+				DeviceMic2Jack "Front Mic Jack"
+				DeviceMic2Priority 300
+			}
+		}
+	}
+}
+
+# Macro HDACaptureDevice - Create the SectionDevice for HDA inputs
+#
+# Arguments:
+# DevName - Name of the device (used as SectionDevice."${var:__DevName}"
+# MicName - Name of the Microphone
+# DevComment - Long name of the SectionDevice
+# JackName - Jack control if available
+# Priority - CapturePriority
+# ConflictCondition1 - (string) if not empty, set ConflictingDevice 1
+# ConflictDev1 - Name of the ConflictingDevice 1
+# ConflictCondition2 - (string) if not empty, set ConflictingDevice 2
+# ConflictDev2 - Name of the ConflictingDevice 2
+# ConflictCondition3 - (string) if not empty, set ConflictingDevice 3
+# ConflictDev3 - Name of the ConflictingDevice 3
+DefineMacro.HDACaptureDevice.SectionDevice."${var:__DevName}" {
+	Comment "${var:__DevComment}"
+
+	If.micsetup {
+		Condition {
+			Type String
+			Empty "${var:SourceControl}"
+		}
+		False.EnableSequence [
+			cset "name='${var:SourceControl}' '${var:__MicName}'"
+		]
+	}
+
+	Value {
+		CapturePriority "${var:__Priority}"
+		Include.value.File "/HDA/HDA-Capture-value.conf"
+		CaptureMasterElem "${var:__MicName} Boost"
+	}
+	If.jack {
+		Condition {
+			Type String
+			Empty "${var:__JackName}"
+		}
+		False.Value.JackControl "${var:__JackName}"
+	}
+
+	If.conflict1 {
+		Condition {
+			Type String
+			Empty "${var:__ConflictCondition1}"
```

Comment:
Use `Empty "${var:-__ConflictCondition1}"` here and remove `ConflictCondition3 "" ConflictDev3 ""` lines from callers. The minus means empty string if the variable does not exists.